metrics/internal/cache/rbd-mirror.go

```diff
@@ -145,31 +137,34 @@ func (s *RBDMirrorStore) WithRBDCommandInput(namespace string) error {
 	return nil
 }
 
-func initCeph(kubeclient clientset.Interface, allowedNamespaces []string) (cephMonitorConfig, error) {
+func initCeph(kubeclient clientset.Interface, cephClusterNamespace, cephAuthNamespace string) (cephMonitorConfig, error) {
 	var err error
 	var namespace string
 	var secret *corev1.Secret
-	// find a namespace which has the expected secret
-	for _, currentNamespace := range allowedNamespaces {
-		secret, err = kubeclient.CoreV1().Secrets(currentNamespace).Get(context.TODO(), "rook-ceph-mon", v1.GetOptions{})
-		// if we are successful, collect the namespace and break
-		if err == nil {
-			namespace = currentNamespace
-			break
+
+	secret, err = kubeclient.CoreV1().Secrets(cephClusterNamespace).Get(context.TODO(), "ocs-metrics-exporter-ceph-auth", metav1.GetOptions{})
+	if err != nil && !apierror.IsNotFound(err) {
+		return cephMonitorConfig{}, err
+	}
+
+	if apierror.IsNotFound(err) {
+		secret, err = kubeclient.CoreV1().Secrets(cephClusterNamespace).Get(context.TODO(), "rook-csi-rbd-provisioner", metav1.GetOptions{})
+		if err != nil {
+			return cephMonitorConfig{}, err
 		}
 	}
-	// under any of these circumstances we should not proceed
-	if err != nil || secret == nil || namespace == "" {
-		return cephMonitorConfig{}, fmt.Errorf("failed to get secret in any of these namespaces %+v: %v",
-			allowedNamespaces, err)
+
+	id, ok := secret.Data["userID"]
+	if !ok {
+		return cephMonitorConfig{}, fmt.Errorf("failed to get ceph user id in namespace %q", namespace)
```

Comment:
You might need to log the secret name here as well?

Reply:
Avoiding secret name in the log. I think that message is enough to hint what is wrong.